### DISCUSSION TITLE
Required fields in repeater interface

### DISCUSSION THREAD

**Discussion by enzolupia** [directus/directus#12296] [Archive]
Hi everyone, I'm trying to use the repeater interface and I need to mark some fields as mandatory but I don't see any option to do that.

I also have a custom interface containing some select dropdowns (v-select) and mandatory properties seems not to work at all.

Is there any way I can get this to work?

Thanks.

**Comment by rijkvanzanten** [comment]
Directus doesn't currently have a way to validate the fields inside a JSON blob, as it's just an arbitrary JSON blob. You can enable validation by using a relational repeater instead 👍🏻

**Reply by enzolupia** [reply]
Hi @rijkvanzanten  thanks for your response.
Is the relational repeater already available on the latest version of directus?

**Reply by rijkvanzanten** [reply]
Yup! The many-to-many and one-to-many interfaces have the same ux/ui as the JSON repeater 👍🏻

**Reply by enzolupia** [reply]
Thanks, I will need to stick with the custom interface since I have some custom logics that I cannot replicate with relational fields.
Assuming I can manage the validation of the fields in the custom interface, do you think Is there any way to disabile the repeater'drawer save button until I get a valid input from the users?

**Reply by rijkvanzanten** [reply]
If you build a custom interface, you're adding the drawer logic in the custom interface, so you can make it do whatever you want 🙂

**Reply by enzolupia** [reply]
Well, that's not the case, I'm not re-implementing the whole repeater interface, I only created a custom interface for JSON type fields and I'm using it inside the repeater interface.
The drawer logic is not part of my interface.

It would be great, from an extensibility POV to allow such interactions.

**Reply by rijkvanzanten** [reply]
Ah gotcha. In that case, there's currently no way to override the parent interface drawer behavior 👍🏻
